Update Changelog for v5.0.0

sledgehammer999 opened the pull request:
Draft for reviewing the changelog for the upcoming RC1 release.
I have left the commit hashes alongside each entry for quicker review.

Let me know if something shouldn't be in the changelog. Or if I have forgotten something. Or if some entry is already backported to v4.6.x.

PS: I mention only entries that make sense to the end user and not internal refactoring.

thalieht commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - FEATURE: Add seeding limits to RSS and Watched folders options UI (glassez) d652a10495fdd5175242f029a7e2b9a6373ea985
+    - BUGFIX: Display error message when unrecoverable error occurred (glassez) a0e41a11dec9e195455d990fd0e87390882c703d
+    - BUGFIX: Update size of selected files when selection is changed (glassez) eddfaedc4d34cfcaf8a38210e6700ace0886cc9f
+    - BUGFIX: Specify a locale if none is set (Chocobo1) fc56fdade75610fabf42426cdc8960b9f1ba77a4
```
IIRC this is only for linux.

thalieht commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - BUGFIX: Update size of selected files when selection is changed (glassez) eddfaedc4d34cfcaf8a38210e6700ace0886cc9f
+    - BUGFIX: Specify a locale if none is set (Chocobo1) fc56fdade75610fabf42426cdc8960b9f1ba77a4
+    - WEBUI: Improve WebUI responsiveness (Chocobo1) 2edb1a0765dc501bb66d133d1bc8ac3c37704f51
+    - WEBUI: Add I2P settings to WebUI (thalieht) a3888811f6ad5a82119805378730a68402765bfa
```
It's in 4.6

thalieht commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - FEATURE: Expose `DHT bootstrap nodes` setting (Chocobo1) dcba9eda009fcb74e8aebbee4a4c5df7248abf06
+    - FEATURE: Improve GUI behavior when adding multiple torrents (glassez) 70b438e6d9249bacf8589ba50d16bbfd617fd199
+    - FEATURE: Add support for [Mark-of-the-Web](https://redcanary.com/threat-detection-report/techniques/mark-of-the-web-bypass/) (Chocobo1) 34f7b75f12390454ed831037ff34531b90307c55
+    - FEATURE: Keep unwanted files in separate folder (configurable) (glassez) e216a9cf53ae1203d148a013ab5d71d6672fe97a
```
I'm afraid `(configurable)` will give the wrong impression that it's location can be changed.

sledgehammer999 replied:
I'll drop `(configurable)` and let them discover the setting on their own.

glassez replied:
Maybe `Allow to keep unwanted files in separate folder`?

glassez commented:
Keep in mind that I have tried to make sure that all PRs related to the 5.0 branch have a corresponding milestone assigned, while the backported ones have a different ones. So it makes sense to compare this Changelog with the list of PRs having 5.0 milestone assigned.

glassez commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - BUGFIX: Specify a locale if none is set (Chocobo1) fc56fdade75610fabf42426cdc8960b9f1ba77a4
+    - WEBUI: Improve WebUI responsiveness (Chocobo1) 2edb1a0765dc501bb66d133d1bc8ac3c37704f51
+    - WEBUI: Add I2P settings to WebUI (thalieht) a3888811f6ad5a82119805378730a68402765bfa
+    - WEBUI: Do not exit the app when WebUI hass failed to start (Hanabishi) 35e4b35d125983051a25173d595119d99b7bda3a
```
```suggestion
    - WEBUI: Do not exit the app when WebUI has failed to start (Hanabishi) 35e4b35d125983051a25173d595119d99b7bda3a
```

glassez commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - WEBUI: Allow to remember torrent content files deletion (David) e69f85782881a4b831ed97694263f027ecfa568d
+    - WEBUI: Leave the fields empty when value is invalid (Chocobo1) 54bcabeca4292a72649202a03eaf09e711d73395
+    - WEBUI: Use natural sorting (Chocobo1) 00372dd559dedd19e3bcb66db8e7d5faf597d4b2
+    - WEBUI: WebAPI: Get torrent count  (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
```
`WEBUI: WebAPI:` looks ugly.

sledgehammer999 replied:
Then maybe it's time I use `WEBAPI` as a category

glassez commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - RSS: Show RSS feed title in HTML browser (Jay) f3b688217eb518ca67de23edd716278efde0b990
+    - SEARCH: Allow users to specify Python executable path (Chocobo1) b3fda76027859707b57530a1de0e2c68c0b27d9f
+    - WINDOWS: Use Fusion style on Windows 10+. It has better compatibility with dark mode (glassez) 5a334175a8b5047cf56552c0bf5a54c3e3e14893
+    - WINDOWS: Improve .torrent association handling (Chocobo1) cba433823ff46c5c799e8d9e7c6dd65c9e8bdd4c
```
As far as I remember, it doesn't make sense anymore, since `WINDOWS: Allow to set qBittorrent as default program`.

glassez commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+    - LINUX: Support for localized man pages (Victor Chernyakin) 2dc1a7d66f246faee94f08c1a033f107ae126f0f
+    - MACOS: Display download/upload speed in dock icon (Nick Korotysh) e6ec3d0c2bbdda2d3228558907353f725a4d45b7
+    - MACOS: Add support for quarantine on macOS (Chocobo1) 98576dacae1ea6fb5369f4aef2cf91426766cb57
+    - OTHER: Switch to C++20
```
I believe this and the following one correlate with `FEATURE: Drop support for Qt5, qmake, autotools, Windows < 10, Windows 32-bit`.

sledgehammer999 replied:
IMO, I prefer to be a tad bit explicit on what is dropped.

glassez replied:
My comment is mostly about different categories used for similar kind of changes.

glassez commented on Changelog:
```diff
@@ -1,3 +1,45 @@
+Unreleased - sledgehammer999 <sledgehammer999@qbittorrent.org> - v5.0.0
+    - FEATURE: Drop support for Qt5, qmake, autotools, Windows < 10, Windows 32-bit
```
It looks strange to me that such changes are mentioned as a FEATURE.

sledgehammer999 replied:
Will be downgraded to `OTHER`

glassez commented:
> Keep in mind that I have tried to make sure that all PRs related to the 5.0 branch have a corresponding milestone assigned, while the backported ones have a different ones. So it makes sense to compare this Changelog with the list of PRs having 5.0 milestone assigned.

#19333 Change default power management to Gnome Session Manager
#19404 Remove column span
#19633 Revamp tracker list widget
#19801 Delay subsequent requests to the same host 
#19957 Follow the parent category options 
#19969 Fix JS memory leak
#20028 Add a class to represent a tag
#20207 Revise conditional for when to use QCollator
#20283 Change "metadata received" stop condition behavior 

This is a list of those changes that should also be mentioned (perhaps by changing the titles to more suitable for showing to the end user).

sledgehammer999 commented:
I had reviewed most of these and the reasons I have not included them:

>https://github.com/qbittorrent/qBittorrent/pull/19333 Change default power management to Gnome Session Manager

I don't think this changes much from a user POV. It seems like a code cleanup mostly.

>https://github.com/qbittorrent/qBittorrent/pull/19404 Remove column span

Backported with c7dfc1ded2c60f7f5749eda579eb460ee3b3ac37

>https://github.com/qbittorrent/qBittorrent/pull/19633 Revamp tracker list widget

Kinda folded into the entry `FEATURE: Improve tracker entries handling`

>https://github.com/qbittorrent/qBittorrent/pull/19801 Delay subsequent requests to the same host

This might be legit. Needs better wording for end user.

>https://github.com/qbittorrent/qBittorrent/pull/19957 Follow the parent category options

OK, now that I opened the PR I realize that is a feature implementation, not an internal bug fix.

>https://github.com/qbittorrent/qBittorrent/pull/19969 Fix JS memory leak

Backported with 081eace057bfb1d18e375f4d224f04f8ee691ddc

>https://github.com/qbittorrent/qBittorrent/pull/20028 Add a class to represent a tag

From the title it is clearly internal code refactoring. However, it actually fixes a bug aka #19744, so I'll mention that.

>https://github.com/qbittorrent/qBittorrent/pull/20207 Revise conditional for when to use QCollator

IMO, meaningless for end user.

>https://github.com/qbittorrent/qBittorrent/pull/20283 Change "metadata received" stop condition behavior

Do you think it is relevant to end users?

glassez commented:
> > #20207 Revise conditional for when to use QCollator
> 
> IMO, meaningless for end user.
> 
> > #20283 Change "metadata received" stop condition behavior
> 
> Do you think it is relevant to end users?

They both fix the problems reported by end-users.

glassez commented:
> Draft for reviewing the changelog for the upcoming RC1 release.

Do you intend to omit the alpha/beta releases?
I would provide at least one beta.

sledgehammer999 commented:
>Do you intend to omit the alpha/beta releases?
I would provide at least one beta.

Sorry, I didn't communicate this well.
With this release cycle I don't want to fumble between alpha/beta/RC designators. I want to choose one designator and monotonically increase it (eg `beta1`, `beta2`).
I probably should choose `beta` as the middle ground. `alpha` indicates very early dev release ("here be dragons") and `RC` indicates very stable, almost ready for final release.

glassez commented:
> `alpha` indicates very early dev release ("here be dragons") and `RC` indicates very stable, almost ready for final release.

I agree that there is not much point in releasing an alpha version (we can consider builds from GitHub as intended for alpha testing). But it makes sense to distinguish between beta and RC releases. I would still provide at least one RR, which should mark the creation of v5.0 branch and so-called "feature/strings freeze".

glassez commented:
@sledgehammer999 
BTW, some bugs are fixed indirectly by fixing `libtorrent`. E.g. #20006.
Shouldn't we mention such fixups in any way?
The Changelog doesn't seem to be the right place to do this. But in the announcement on the website, they could be mentioned to notify users.

sledgehammer999 commented:
I pushed the suggested changes as a separate commit to ease review.


>which should mark the creation of v5.0 branch and so-called "feature/strings freeze".

👍 

> > > #20207 Revise conditional for when to use QCollator
> > 
> > 
> > IMO, meaningless for end user.
> > > #20283 Change "metadata received" stop condition behavior
> > 
> > 
> > Do you think it is relevant to end users?
> 
> They both fix the problems reported by end-users.

I propose that those 2 PRs to be backported instead. What do you think?

glassez commented:
> I propose that those 2 PRs to be backported instead. What do you think?

I have no serious objections. However, it is worth pointing out that #20283 affects strings, so you will have to make sure that translators have the opportunity to take care of them.

glassez commented on Changelog:
```diff
@@ -8,36 +7,38 @@ Unreleased - sledgehammer999 <sledgehammer999@qbittorrent.org> - v5.0.0
-    - WEBUI: WebAPI: Allow WebAPI to specify filename and mime type for result data  (glassez) 60bb819e2e839d691dec6915d592d4a2c66d1de2
+    - WEBAPI: Get torrent count  (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
+    - WEBAPI: Fix wrong timestamp values  (Chocobo1) 114652205c92bf7fcf63f355cab439a5eb4b184f
+    - WEBAPI: Allow WebAPI to specify filename and mime type for result data  (glassez) 60bb819e2e839d691dec6915d592d4a2c66d1de2
```
From API users perspective it should look like: `WEBAPI: Send binary data with filename and mime type specified`

glassez commented on Changelog:
```diff
@@ -8,36 +7,38 @@ Unreleased - sledgehammer999 <sledgehammer999@qbittorrent.org> - v5.0.0
-    - WEBUI: WebAPI: Get torrent count  (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
-    - WEBUI: WebAPI: Fix wrong timestamp values  (Chocobo1) 114652205c92bf7fcf63f355cab439a5eb4b184f
-    - WEBUI: WebAPI: Allow WebAPI to specify filename and mime type for result data  (glassez) 60bb819e2e839d691dec6915d592d4a2c66d1de2
+    - WEBAPI: Get torrent count  (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
```
```suggestion
    - WEBAPI: Add endpoint to get torrent count  (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
```

glassez commented on Changelog:
```diff
@@ -8,36 +7,38 @@ Unreleased - sledgehammer999 <sledgehammer999@qbittorrent.org> - v5.0.0
+    - WEBAPI: Fix wrong timestamp values  (Chocobo1) 114652205c92bf7fcf63f355cab439a5eb4b184f
+    - WEBAPI: Allow WebAPI to specify filename and mime type for result data  (glassez) 60bb819e2e839d691dec6915d592d4a2c66d1de2
     - RSS: Show RSS feed title in HTML browser (Jay) f3b688217eb518ca67de23edd716278efde0b990
+    - RSS: Rate limit requests to the same host (jNullj) c5d7b62473887c337c249ac7b3bd6420286c7e83
```
```suggestion
    - RSS: Allow to set delay between requests to the same host (jNullj) c5d7b62473887c337c249ac7b3bd6420286c7e83
```

glassez commented on Changelog:
```diff
@@ -8,36 +7,38 @@ Unreleased - sledgehammer999 <sledgehammer999@qbittorrent.org> - v5.0.0
     - BUGFIX: Display error message when unrecoverable error occurred (glassez) a0e41a11dec9e195455d990fd0e87390882c703d
     - BUGFIX: Update size of selected files when selection is changed (glassez) eddfaedc4d34cfcaf8a38210e6700ace0886cc9f
-    - BUGFIX: Specify a locale if none is set (Chocobo1) fc56fdade75610fabf42426cdc8960b9f1ba77a4
+    - BUGFIX: Tags with leading whitespace can now be removed (glassez) 7a4119259703179909dd37153031f11a3ef617e8
```
It behaves differently. It normalizes tags by trimming leading/trailing spaces.

glassez commented on Changelog:
```diff
@@ -1,3 +1,46 @@
+    - FEATURE: Allow to remove tracker from tracker filter widget menu (glassez) 30189ae943b9a85d39484f6cdc64e2200518654f
+    - FEATURE: Implement `Reannounce In` column (Hanabishi) c394868f87e71d8ab614c6ae5463c712acffd172
+    - FEATURE: Expose `DHT bootstrap nodes` setting (Chocobo1) dcba9eda009fcb74e8aebbee4a4c5df7248abf06
+    - FEATURE: Improve GUI behavior when adding multiple torrents (glassez) 70b438e6d9249bacf8589ba50d16bbfd617fd199
```
It was backported into v4.6 by #19582.

sledgehammer999 commented:
Two new commits. One addresses the comments and the other mentions the new commits from master.
I have left out PR #20465 #20463 #20442 because I am not sure if they'll be backported or not.

Chocobo1 commented:
>I have left out PR https://github.com/qbittorrent/qBittorrent/pull/20465 https://github.com/qbittorrent/qBittorrent/pull/20463 https://github.com/qbittorrent/qBittorrent/pull/20442 because I am not sure if they'll be backported or not.

As for #20465, if it can be cherry-picked cleanly into v4.x then I'm ok with it.

glassez commented on Changelog:
```diff
@@ -1,3 +1,48 @@
+    - FEATURE: Add seeding limits to RSS and Watched folders options UI (glassez) d652a10495fdd5175242f029a7e2b9a6373ea985
+    - FEATURE: Subcategories implicitly follow the parent category options (glassez) d652aa1c78a0455e4787e1fe95b95b767dd1c2667982e10495fdd5175242f029a7e2b9a6373ea985
+    - FEATURE: Add support for SSL torrents (Chocobo1, Radu Carpa) cffd74b62a0e19f9c1e291306125475e24659f86
+    - FEATURE: Option to name each qbittorrent instance (Chocobo1) 46e8ee50c833ac3f0691a05a7d9c4b6dcb095641
```
```suggestion
    - FEATURE: Add option to name each qbittorrent instance (Chocobo1) 46e8ee50c833ac3f0691a05a7d9c4b6dcb095641
```

glassez commented on Changelog:
```diff
@@ -1,3 +1,48 @@
+    - WEBUI: Improve WebUI responsiveness (Chocobo1) 2edb1a0765dc501bb66d133d1bc8ac3c37704f51
+    - WEBUI: Do not exit the app when WebUI has failed to start (Hanabishi) 35e4b35d125983051a25173d595119d99b7bda3a
+    - WEBUI: Add `Moving` filter to side panel (xavier2k6) f067ab169224013cb4bf21630b8a3eaa5d7d2dd6
+    - WEBUI: Add dark theme (d47081) 9d90141c29b80b03631e19cba0fa6f1cddc30423
```
Is it applied to entire WebUI?

sledgehammer999 replied:
I think so. The PR is #19901

glassez commented on Changelog:
```diff
@@ -1,3 +1,48 @@
+    - WINDOWS: Use Fusion style on Windows 10+. It has better compatibility with dark mode (glassez) 5a334175a8b5047cf56552c0bf5a54c3e3e14893
+    - WINDOWS: Allow to set qBittorrent as default program (glassez) 6860c0d60de0ad2bb4374096354fadf3d0261a43
+    - LINUX: Add support for systemd power management (Chocobo1) f3f9cfe44ed49d5bf3de0e318fa9c8ec48645ca4
+    - LINUX: Support for localized man pages (Victor Chernyakin) 2dc1a7d66f246faee94f08c1a033f107ae126f0f
```
```suggestion
    - LINUX: Add support for localized man pages (Victor Chernyakin) 2dc1a7d66f246faee94f08c1a033f107ae126f0f
```

sledgehammer999 commented:
I rebased on master, I addressed the comments and added entries for the new commit.

sledgehammer999 commented on Changelog:
```diff
@@ -1,3 +1,49 @@
+    - WEBUI: Leave the fields empty when value is invalid (Chocobo1) 54bcabeca4292a72649202a03eaf09e711d73395
+    - WEBUI: Use natural sorting (Chocobo1) 00372dd559dedd19e3bcb66db8e7d5faf597d4b2
+    - WEBUI: Improve WebUI login behavior (JayRet) f265eb0166f57a5dd866a5461b7bf2ab3076ee9f
+    - WEBAPI: Add endpoint to get torrent count (glassez) cecbb44a031c96e8fce48f12eeddda47b9c85a27
```
Hmm, I think this was backported in 4.6.1. I will drop it.

glassez replied:
> I think this was backported in 4.6.1

Yes.
#19831

glassez commented:
> If there isn't any other comment, I am going to merge the Changelog sometime in the next ~12 hours.

So can we still leave it open to add new changes as soon as they are merged?

sledgehammer999 commented:
>So can we still leave it open to add new changes as soon as they are merged?

I am not sure what workflow you propose here.
Personally, I wanted to merge this so I could **maybe** do a beta release the weekend.

glassez commented:
> I am not sure what workflow you propose here.

1. PR with changes than need to be mentioned in Changelog is merged
2. Commit that adds new Changelog entry is added here
3. Repeat until release

sledgehammer999 commented:
I have pushed current Changelog, changed version numbers and tagged a beta release. I haven't done the builds yet. I'll do them tomorrow based on the tag.
You can continue pushing into master and maybe create a new draft PR to keep track of Changelog updates.
